src/streamlink/plugins/svtplay.py

```diff
@@ -1,92 +1,146 @@
+
+    author = None
+    category = None
+    title = None
+
+    url_re = re.compile(r'''
+        https?://(?:www\.)?(?:svtplay|oppetarkiv).se(?P<path>.*)
+    ''', re.VERBOSE)
+
+    latest_episode_url_re = re.compile(r'''
+        class="play_titlepage__latest-video"\s+href="(?P<url>[^"]+)"
+    ''', re.VERBOSE)
+
+    live_id_re = re.compile(r'.*/(?P<live_id>[^?]+)')
+
+    vod_id_re = re.compile(r'''
+        (?:DATA_LAKE\s+=\s+{"content":{"id":|data-video-id=)
+        "(?P<vod_id>[^"]+)"
+    ''', re.VERBOSE)
+
+    _video_schema = validate.Schema({
+        validate.optional('programTitle'): validate.text,
+        validate.optional('episodeTitle'): validate.text,
+        'videoReferences': [{
+            'url': validate.url(),
+            'format': validate.text,
+        }],
+        validate.optional('subtitleReferences'): [{
+            'url': validate.url(),
+            'format': validate.text,
+        }],
+    })
+
+    arguments = PluginArguments(
+        PluginArgument(
+            'mux-subtitles',
+            action='store_true',
+            help="Automatically mux available subtitles in to the output stream.",
+        ),
+    )
+
     @classmethod
-    def can_handle_url(self, url):
-        return _url_re.match(url)
+    def can_handle_url(cls, url):
+        return cls.url_re.match(url) is not None
+
+    def get_author(self):
+        if self.author is not None:
+            return self.author
 
-    def _create_streams(self, stream_type, parser, video):
-        try:
-            streams = parser(self.session, video["url"])
-            return streams.items()
-        except IOError as err:
-            self.logger.error("Failed to extract {0} streams: {1}",
-                              stream_type, err)
+    def get_category(self):
+        if self.category is not None:
+            return self.category
+
+    def get_title(self):
+        if self.title is not None:
+            return self.title
 
     def _get_streams(self):
-        # Retrieve URL page and search for new type of video ID
-        res = self.session.http.get(self.url)
-        match = _id_re.search(res.text)
-
-        # Use API if match, otherwise resort to old method
-        if match:
-            vid = match.group("id")
-            res = self.session.http.get(API_URL.format(vid))
-
-            videos = self.session.http.json(res, schema=_video_schema)
-            mapper = StreamMapper(cmp=lambda format, video: video["format"] == format)
-            mapper.map("hls", self._create_streams, "HLS", HLSStream.parse_variant_playlist)
-            mapper.map("hds", self._create_streams, "HDS", HDSStream.parse_manifest)
-        else:
-            res = self.session.http.get(self.url, params=dict(output="json"))
-            videos = self.session.http.json(res, schema=_old_video_schema)
+        path = self.url_re.match(self.url).group('path')
+        log.debug("Path={0}".format(path))
+
+        if path.startswith('/kanaler/'):
```

Comment:
I think it may be better to add the different paths to the regex and check for the capture groups. This way a reader of the plugin code can check for the supported paths much more easily.

Also, please move this into class methods instead of having a large if-else block.